src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -136,21 +136,7 @@ private void Initialize()
                 // machine connecting to public server).
 
                 // SMTP RFC's require ASCII only host names in the HELO/EHLO message.
-
-                // TODO: https://github.com/dotnet/corefx/issues/19605
-                //
-                //   Catching NotSupportedException here because HostName is currently PNSE
-                //   on UapAot. HostName is planned to be restored for .NetCore 2.1
-                //   so once that's done, the try-catch can be removed.
-                string clientDomainRaw;
-                try
-                {
-                    clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
-                }
-                catch (NotSupportedException)
-                {
-                    clientDomainRaw = "LocalHost";
-                }
+                string clientDomainRaw = IPGlobalProperties.GetIPGlobalProperties().HostName;
```

Comment:
I think it can be tested indirectly if you follow the code to see how it is used and what other things change as a result of it. For example, perhaps something on the wire is using that name.  So, the loopback SMTP server could inspect the wire protocol, etc.